core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -845,20 +846,69 @@ object JsonAST {
   }
 
   object RenderSettings {
+    /**
+     * Pretty-print JSON with 2-space indentation.
+     */
     val pretty = RenderSettings(2)
+    /**
+     * Compact print JSON on one line.
+     */
     val compact = RenderSettings(0)
+
+    /**
+     * Ranges of chars that should be escaped if this JSON is to be evaluated
+     * directly as JavaScript (rather than by a valid JSON parser).
+     */
+    val jsEscapeChars =
+      List(('\u00ad', '\u00ad'),
+           ('\u0600', '\u0604'),
+           ('\u070f', '\u070f'),
+           ('\u17b4', '\u17b5'),
+           ('\u200c', '\u200f'),
+           ('\u2028', '\u202f'),
+           ('\u2060', '\u206f'),
+           ('\ufeff', '\ufeff'),
+           ('\ufff0', '\uffff'))
+        .foldLeft(Set[Char]()) {
+          case (set, (start, end)) =>
+            set ++ (start to end).toSet
+        }
+
+    /**
+     * Pretty-print JSON with 2-space indentation and escape all JS-sensitive
+     * characters.
+     */
+    val prettyJs = RenderSettings(2, jsEscapeChars)
+    /**
+     * Compact print JSON on one line and escape all JS-sensitive characters.
+     */
+    val compactJs = RenderSettings(0, jsEscapeChars)
   }
+  /**
+   * RenderSettings allows for customizing how JSON is rendered to a String.
+   * At the moment, you can customize the indentation (if 0, all the JSON is
+   * printed on one line), the characters that should be escaped (in addition
+   * to a base set that will always be escaped for valid JSON), and whether or
+   * not a space should be included after a field name.
+   */
   case class RenderSettings(
     indent: Int,
+    escapeChars: Set[Char] = Set(),
     spaceAfterFieldName: Boolean = false
   ) {
     val lineBreaks_? = indent > 0
   }
 
+  /**
+   * Render `value` using `RenderSettings.pretty`.
```

Comment:
Maybe this should mention it's only for JSON instead of expecting people to click through to the render settings docs? Also, could we make the reference to those settings a link?